plugins/woocommerce/includes/admin/class-wc-admin-notices.php

```diff
@@ -479,7 +479,7 @@ public static function legacy_shipping_notice() {
 	 */
 	public static function no_shipping_methods_notice() {
         // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		if ( wc_shipping_enabled() && ( empty( $_GET['page'] ) || empty( $_GET['tab'] ) || 'wc-settings' !== $_GET['page'] || 'shipping' !== $_GET['tab'] ) ) {
+		if ( wc_shipping_enabled() && ( ! is_wc_admin_settings_page() || empty( $_GET['tab'] ) || 'shipping' !== $_GET['tab'] ) ) {
```

Comment:
Note: same as original, but adds `is_admin()` check, which looks safe based on the usage context.